pkg/ingestor/api/api.go

```diff
@@ -116,6 +131,34 @@ func (g *IngestorAPI) Ingest(_ context.Context, clusterName string, runID string
 	return nil
 }
 
+func (g *IngestorAPI) checkPreviousRun(ctx context.Context, clusterName string, runID string) error {
+	var resNum int64
+	var err error
+	for _, collection := range collections.GetCollections() {
+		db := adapter.MongoDB(g.providers.StoreProvider).Collection(collection)
+		filter := bson.M{
+			"runtime": bson.M{
+				"runID":   runID,
+				"cluster": clusterName,
+			},
+		}
+		resNum, err = db.CountDocuments(ctx, filter, nil)
+		if err != nil {
+			return fmt.Errorf("error counting documents in collection %s: %w", collection, err)
+		}
+		if resNum != 0 {
+			log.I.Infof("Found %d element in collection %s", resNum, collection)
+
+			break
+		}
+	}
+	if resNum != 0 {
+		return fmt.Errorf("ingestion already completed [%s:%s]", clusterName, runID)
+	}
```

Comment:
nit: Any reason not to return in the first `resNum != 0` here?

Reply:
hum not sure. refactoring it.